Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -109,103 +154,117 @@ final class SchemesGenerator: SchemesGenerating {
-                                                           pbxTarget: pbxTarget,
-                                                           projectName: projectName),
-                              launchAction: targetLaunchAction(target: target,
-                                                               pbxTarget: pbxTarget,
-                                                               projectName: projectName),
-                              profileAction: targetProfileAction(target: target,
-                                                                 pbxTarget: pbxTarget,
-                                                                 projectName: projectName),
-                              analyzeAction: targetAnalyzeAction(),
-                              archiveAction: targetArchiveAction())
-        try scheme.write(path: schemePath.path, override: true)
-    }
-
-    /// Generates the scheme test action for a given target.
-    ///
-    /// - Parameters:
-    ///   - target: Target manifest.
-    ///   - pbxTarget: Xcode native target.
-    ///   - projectName: Project name with .xcodeproj extension.
+    ///   - scheme: Scheme manifest.
+    ///   - project: Project manifest.
+    ///   - generatedProject: Generated Xcode project.
     /// - Returns: Scheme test action.
-    func targetTestAction(target: Target,
-                          pbxTarget: PBXNativeTarget,
-                          projectName: String) -> XCScheme.TestAction? {
+    func schemeTestAction(scheme: Scheme,
+                          project: Project,
+                          generatedProject: GeneratedProject) -> XCScheme.TestAction? {
         var testables: [XCScheme.TestableReference] = []
-        if target.product.testsBundle {
-            let reference = targetBuildableReference(target: target,
-                                                     pbxTarget: pbxTarget,
-                                                     projectName: projectName)
-            let testable = XCScheme.TestableReference(skipped: false,
-                                                      buildableReference: reference)
+        var preActions: [XCScheme.ExecutionAction] = []
+        var postActions: [XCScheme.ExecutionAction] = []
+
+        scheme.testAction?.targets.forEach { name in
+            guard let target = project.targets.first(where: { $0.name == name }), target.product.testsBundle else { return }
+            guard let pbxTarget = generatedProject.targets[name] else { return }
+            
+            let reference = self.targetBuildableReference(target: target,
+                                                          pbxTarget: pbxTarget,
+                                                          projectName: generatedProject.name)
+            
+            let testable = XCScheme.TestableReference(skipped: false, buildableReference: reference)
             testables.append(testable)
```

Comment:
Generally if there is a loop and inside the closure for the loop contents are being appended to an array it might make a good use-case for the `map` function. Maybe worth considering here - it's a very minor point so feel free to ignore it.

Reply:
As you see, I'm using `guard let` above therefore using `map` will require return `nil` then filter nil in the end.

I also want to keep coding style as before.